### ISSUE TITLE
Error with "#if WIN32" in pcd_grabber.cpp?

### ISSUE THREAD

**Issue by FSund** [PointCloudLibrary/pcl#3667]
I think this is not valid code if WIN32 is not defined: 
```
#if WIN32
```
https://github.com/PointCloudLibrary/pcl/blob/master/io/src/pcd_grabber.cpp#L207

Should it be changed to an `ifdef` instead?

For context: I'm trying to compile PCL on Windows using MSYS2, and I got the following error:
```
C:/msys64/home/filip.sund/code/pcl/io/src/pcd_grabber.cpp:207:10: error: #if with no expression
  207 | #if WIN32
      |          ^
```

Since I'm using MSYS2 it means I'm only kind-of on WIN32. I think? I'm not sure of the exact details here, unfortunately. I've tried changing it to an `ifdef`, and it seems to compile fine.

**Comment by kunaltyagi** [comment]
Please go ahead and create a PR. If possible, you could find other WIN32 and check if they are being tested properly (`#if defined(WIN32) && condition_on_WIN32` for complicated checks, `#ifdef` for simple ones)

**Comment by FSund** [comment]
I have attempted a fix in #3668. I touched `_WIN32` occurrences in addition to `WIN32`, hope that is okay.
